Runtime log level selection for linux platform

sgunes-wirepas opened the pull request:
- Introduce the OVERRIDE_MAX_LOG_LEVEL parameter that can be passed as a compiler flag to override the MAX_LOG_LEVEL value for each logging module.
- Implement the Platform_set_log_level function for the linux platform to dynamically set the current maximum log level. When targeting the linux platform, OVERRIDE_MAX_LOG_LEVEL can be set to DEBUG_LOG_LEVEL to support dynamically setting the log level up to debug level.

StephaneTriomphe commented on lib/platform/linux/logger.c:
```diff
@@ -69,10 +74,46 @@ static inline void print_prefix(char level, char * module)
+    }
+}
+
+int Platform_set_log_level(const int level)
```
I don't know what was the spec for this modification. But I'd like that this dynamic log level could target one file only. Increasing the level to Debug for all modules will generate too much unwanted logs.
And this level should also be saved, so that on reset, it restarts with the requested log level. Init phase logs are very important.

GwendalRaoul replied:
It would be a much bigger change as log system do not really have semantic for module id for now. Each file select its own log prefix.
But for sure it would be nice, even if at run time it could be hard to use.

sgunes-wirepas replied:
This was mainly for the linux gateway to help with testing. In linux gateway sink service the log level can be set in the beginning of the main function based on an environment variable. I agree there are too many logs when debug is selected. I was thinking about adding a new "trace" level and move some of the logs to use that.

I think module level granularity becomes difficult because of the configuration part. Would we need some configuration file to define all the levels for each module? There's the [zlog](https://github.com/HardySimpson/zlog) library that looks like it has module("category") specific level selection out of the box, but I don't know if it would be worth taking something like that into use?

sgunes-wirepas committed:
Runtime log level selection interface and implementation for linux

This change introduces runtime log level selection globally or per
logging module. Applications can use Platform_set_log_level or
Platform_set_module_log_level functions to set log levels during
runtime.

Each logging module uses a pointer to the log level managed by the
platform implementation to minimize runtime checks.

It is possible to define the NO_BUILD_ALL_LOG_LEVELS parameter to
prevent building of logging functions with higher levels than the
MAX_LOG_LEVEL parameter, like before.

sgunes-wirepas committed:
Disable logging in tests
